CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+# Contributing to opentelemetry-go-compile-instrumentation
+
+The go compile instrumentation SIG meets regularly. See the
+OpenTelemetry
+[community](https://github.com/open-telemetry/community?tab=readme-ov-file#implementation-sigs)
+repo for information on this and other SIGs.
+
+See the [public meeting
+notes](https://docs.google.com/document/d/1XkVahJfhf482d3WVHsvUUDaGzHc8TO3sqQlSS80mpGY/edit)
+for a summary description of past meetings. You can also get in touch on slack channel 
+[#otel-go-compt-instr-sig](https://cloud-native.slack.com/archives/C088D8GSSSF)
+
+## Development
+
+TBD
+
+## Pull Requests
+
+### How to Send Pull Requests
+
+Everyone is welcome to contribute code to `opentelemetry-go-compile-instrumentation` via
+GitHub pull requests (PRs).
+
+To create a new PR, fork the project in GitHub and clone the upstream
+repo:
+
+```sh
+git clone https://github.com/open-telemetry/opentelemetry-go-compile-instrumentation
+```
+
+This would put the project in the `opentelemetry-go-compile-instrumentation` directory in
+current working directory.
+
+Enter the newly created directory and add your fork as a new remote:
+
+```sh
+git remote add <YOUR_FORK> git@github.com:<YOUR_GITHUB_USERNAME>/opentelemetry-go-compile-instrumentation
+```
+
+Check out a new branch, make modifications, run linters and tests, update
+`CHANGELOG.md`, and push the branch to your fork:
+
+```sh
+git checkout -b <YOUR_BRANCH_NAME>
+# edit files
+# update changelog
+git add -p
+git commit
+git push <YOUR_FORK> <YOUR_BRANCH_NAME>
+```
+
+Open a pull request against the main `opentelemetry-go-compile-instrumentation` repo. Be sure to add the pull
+request ID to the entry you added to `CHANGELOG.md`.
+
+Avoid rebasing and force-pushing to your branch to facilitate reviewing the pull request.
+Rewriting Git history makes it difficult to keep track of iterations during code review.
+All pull requests are squashed to a single commit upon merge to `main`.
+
+### How to Receive Comments
+
+* If the PR is not ready for review, please put `[WIP]` in the title,
+  tag it as `work-in-progress`, or mark it as
+  [`draft`](https://github.blog/2019-02-14-introducing-draft-pull-requests/).
+* Make sure CLA is signed and CI is clear.
+
+### How to Get PRs Merged
+
+A PR is considered **ready to merge** when:
+
+* It has received two qualified approvals[^1].
```

Comment:
Based on my past experience, I think it makes sense to start with 1 approval (so you optimize agility over stability), as the project gets more mature (e.g. reaching release candidate or stable), switch to 2 approvals.

Reply:
Good point. +1 to your suggestion.

Reply:
@open-telemetry/go-compile-instrumentation-maintainers What do you think?

Reply:
Sounds good to me. I think most SIGs follow a similar policy.